adapter/src/main/java/com/twilio/kudu/sql/mutation/CubeMutationState.java

```diff
@@ -18,30 +18,35 @@
 import org.apache.calcite.util.Pair;
 import org.apache.kudu.Type;
 import org.apache.kudu.client.KuduException;
+import org.apache.kudu.client.KuduUtil;
+import org.apache.kudu.client.OperationResponse;
 import org.apache.kudu.client.PartialRow;
 import org.apache.kudu.client.Upsert;
+import org.apache.kudu.util.ByteVec;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.math.BigDecimal;
 import java.util.HashMap;
 import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
 public class CubeMutationState extends MutationState {
 
   private static final Logger logger = LoggerFactory.getLogger(CubeMutationState.class);
 
-  // map from dimensions (pk columns of cube table) to measures (non pk columns)
-  // the inner map is from column index to column value
-  // the event time column value is truncated to the cube time rollup
-  private final Map<Map<Integer, Object>, Map<Integer, Object>> aggregatedValues = new HashMap<>();
+  // map from dimensions (pk columns of cube table encoded as a byte[]) to
+  // measures (non pk columns)
+  // the event time pk column value is truncated to the cube time rollup
+  private final Map<ByteVec, Object[]> aggregatedValues = new HashMap<>();
```

Comment:
Represent the primary key of the cube table as an encoded byte[] (using Kudu's encoding).